src/PluginsManager/Microsoft.Performance.Toolkit.PluginsManager.Core/Installation/PluginInstaller.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Performance.SDK;
+using Microsoft.Performance.Toolkit.PluginsManager.Core.Concurrency;
+using Microsoft.Performance.Toolkit.PluginsManager.Core.Packaging;
+using Microsoft.Performance.Toolkit.PluginsManager.Core.Registry;
+using Microsoft.Performance.Toolkit.PluginsManager.Core.Utils;
+
+namespace Microsoft.Performance.Toolkit.PluginsManager.Core.Installation
+{
+    // TODO: #254 docstrings
+    public sealed class PluginInstaller
+    {
+        private readonly PluginRegistry pluginRegistry;
+
+        public PluginInstaller(PluginRegistry pluginRegistry)
+        {
+            this.pluginRegistry = pluginRegistry;
+        }
+
+        public async Task<IReadOnlyCollection<InstalledPlugin>> GetAllInstalledPluginsAsync(
+            CancellationToken cancellationToken)
+        {
+            using (this.pluginRegistry.UseLock(cancellationToken))
+            {
+                return await this.pluginRegistry.GetInstalledPlugins();
+            }  
+        }
+
+        public async Task<bool> InstallPluginAsync(
+            PluginPackage pluginPackage,
+            string installationDir,
+            Uri sourceUri,
+            CancellationToken cancellationToken,
+            IProgress<int> progress)
+        {
+            using (this.pluginRegistry.UseLock(cancellationToken))
+            {
+                InstalledPlugin installedPlugin = await InstallPluginPackageCoreAsync(
+                    pluginPackage,
+                    installationDir,
+                    sourceUri,
+                    cancellationToken,
+                    progress);
```

Comment:
Might want to check cancellation token here, and mark the installed dir as to delete if cancelled